lib/Site.js

```diff
@@ -356,22 +357,55 @@ Site.prototype.generatePages = function () {
   // Render the final rendered page to the output folder.
   const { baseUrl } = this.siteConfig;
   const builtFiles = {};
-  const pages = this.siteConfig.pages || [];
   const processingFiles = [];
-  this.pageModels = pages.map(page => this.createPageData({
-    baseUrl,
-    pageSrc: page.src,
-    title: page.title,
-    pageTemplate: this.pageTemplate,
-  }));
-  this.pageModels.forEach((page) => {
-    processingFiles.push(page.generate(builtFiles));
-  });
-  return new Promise((resolve, reject) => {
-    Promise.all(processingFiles)
-      .then(resolve)
-      .catch(reject);
+  return this.buildSiteData().then((frontMatterData) => {
+    const pages = frontMatterData || [];
+    this.pageModels = pages.map(page => this.createPageData({
+      baseUrl,
+      pageSrc: page.src,
+      title: page.title,
+      pageTemplate: this.pageTemplate,
+    }));
+    this.pageModels.forEach((page) => {
+      processingFiles.push(page.generate(builtFiles));
+    });
+    return new Promise((resolve, reject) => {
+      Promise.all(processingFiles)
+        .then(resolve)
+        .catch(reject);
+    });
+  },
+  );
+};
+
+/**
+ * Builds the siteData file that specifies addressable pages and their associated keywords
+ */
+Site.prototype.buildSiteData = function () {
+  const filePaths = walkSync(this.rootPath, { directories: false })
+    .filter(currentPath => currentPath.endsWith('.md') && !currentPath.startsWith('_site'));
+  const frontMatterData = [];
+  const promises = [];
+  filePaths.forEach((currentFile) => {
+    promises.push(
+      new Promise((resolve, reject) => {
```

Comment:
If we are using `fs-extra-promise`, we can use `fs.readFileAync(path).then...` ?

Reply:
Yeah that's true. I'll switch the methods.